### PR TITLE
add SeedNodes to CConnman::Options

### PR DESCRIPTION
**Pull request by benma**
Start of a series of changes to clean up the instantiation of connman
by decoupling the command line arguments.

### CODE REVIEW

**Comment by benma** [comment]
Pinging @theuni

**Comment by benma** [comment]
@fanquake I don't think the P2P label is accurate. It should be `Refactoring`, like #9897, which this PR is a subset of.

File: src/net.cpp [hunk 1]
```diff
@@ -2212,6 +2212,10 @@ bool CConnman::Start(CScheduler& scheduler, std::string& strNodeError, Options c
 
     SetBestHeight(connOptions.nBestHeight);
 
+    BOOST_FOREACH(const std::string& strDest, connOptions.vSeedNodes) {
```
Comment (fanquake) [inline src/net.cpp]: Can you change this to not introduce new Boost code.

File: src/init.cpp [hunk 2]
```diff
@@ -1659,6 +1654,11 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
     connOptions.nMaxOutboundTimeframe = nMaxOutboundTimeframe;
     connOptions.nMaxOutboundLimit = nMaxOutboundLimit;
 
+    if (gArgs.IsArgSet("-seednode")) {
+        BOOST_FOREACH(const std::string& strDest, gArgs.GetArgs("-seednode"))
```
Comment (fanquake) [inline src/init.cpp]: Even though your only moving this, if we're going to touch these lines, can you remove Boost.

File: src/net.cpp (hunk 1 above)
Comment (benma) [inline src/net.cpp]: Done

File: src/init.cpp (hunk 2 above)
Comment (benma) [inline src/init.cpp]: Done

**Review by theuni — APPROVED** [review]
Nice. Ridding CConnman of the argument parsing has been on my todo for a long time.

utACK 5d67526026cd41845e06e93e5ae216bc1a7b9596

**Pull request merged**